perf/bloom_filter/bloom_filter.cc

```diff
@@ -0,0 +1,297 @@
+	resultsFile.open(resultsFileName, std::ios_base::in);
+	if (!resultsFile) {
+		std::cout << "Results file "
+			  << resultsFileName
+			  << " does not exist. Creating it." << std::endl;
+		resultsFile.open(resultsFileName, std::ios_base::app);
+		resultsFile << "builds:" << std::endl;
+	} else {
+		std::cout << "Results file "
+			  << resultsFileName
+			  << " does exist. Starting to write." << std::endl;
+		resultsFile.close();
+		resultsFile.open(resultsFileName, std::ios_base::app);
+	}
+	resultsFile << "    - build:" << std::endl;
+	resultsFile << "            build_name: " << build_name << std::endl;
+	resultsFile << "            tuple_count: " << tuple_count << std::endl;
+	resultsFile << "            spaces:" << std::endl;
+	return 0;
+}
+
+int
+stop(box_function_ctx_t *ctx, const char *args, const char *args_end)
+{
+	(void)ctx;
+	(void)args;
+	(void)args_end;
+
+	delete data;
+	delete nonexistenetTuple;
+	resultsFile.close();
+
+	std::cout << "FINISHED benchmarking." << std::endl;
+	std::cout << "-----------------------------------------------------------------" << std::endl;
+	std::cout << std::endl;
+	return 0;
+}
+
+static void
+print_result(uint32_t tuple_count, const struct BenchAlgorithmResult &result)
+{
+	resultsFile.precision(1);
+	resultsFile << "                    - space:" << std::endl;
+	resultsFile << "                            space_name: " << std::fixed
+		    << result.space_name << std::endl;
+	resultsFile << "                            insert: "
+		    << 1000 * tuple_count / result.insertTimeMs.elapsed_time << std::endl;
+	resultsFile << "                            select: "
+		    << 1000 * tuple_count / result.selectTimeMs.elapsed_time << std::endl;
+	resultsFile << "                            bloom_size: "
+		    << result.bloom_bsize << std::endl;
+	std::cout << "Written new result for " << result.space_name
+		  << " space to the disk." << std::endl;
+	std::cout << std::endl;
+}
+
+/**
+ * Function that fills data for future queries.
+ * Also warms up the hole db.
+ */
+static void bench_setup(uint32_t space_id, TupleData *data, BenchAlgorithmResult &result)
+{
+	PerfTimer timer{};
+	std::cout << "Started inserting tuples into "
+		  << result.space_name << " space." << std::endl;
+	timer.start();
+
+	for (const auto & t : data->insertTuples) {
+		box_insert(space_id, t.begin, t.end, NULL);
+	}
+	timer.stop();
+
+	result.insertTimeMs = BenchTimeResult{timer.elapsed_ms()};
+	std::cout << "Finished inserting tuples into "
+	          << result.space_name << " space." << std::endl;
+}
+
+static void
+bench_select(uint32_t space_id, int index_id, TupleData *data,
+			 BenchAlgorithmResult &result)
+{
+	PerfTimer timer{};
+	timer.start();
+	for (const auto & t: data->selectTuples) {
+		box_tuple_t *tuple;
+		int res = box_index_get(space_id, index_id, t.index_begin, t.index_end, &tuple);
+		if (res == -1) {
+			std::cout << "Error occurred while executing select query." << std::endl;
+			exit(1);
+		}
```

Comment:
Please replace this with an assertion, this can actually slow down your benchmark. And add a similar check for `insert`.

Reply:
Done (commented error reporting variant)

Reply:
Get the answer.